src/aspire/abinitio/commonline_base.py

```diff
@@ -167,6 +167,10 @@ def build_clmatrix(self):
         # Note that only use half of each ray
         pf = self._apply_filter_and_norm("ijk, k -> ijk", pf, r_max, h)
 
+        # Setup a progress bar
+        _total_pairs_to_test = self.n_check * (self.n_check - 1) // 2
```

Comment:
This is only true for when `self.n_check = self.n_img`, ie. in the default case `n_check=None`. In the case where `n_check` is actually used (if this case ever exists) I believe this should be `self.n_img * (self.n_check - 1) // 2`.

Another thing to possibly consider is shifts. The most inner loop is over shifts. If you wanted to include that in the pbar you'd want to multiply by `len(shifts)`. That might be overkill though.

Reply:
Re the first remark.  Yeah, I think you are right, I misunderstood what `n_remaining` was. Changed, thanks.

Re the second point, so long as the shifts are a constant (ie scales iteration linearly)  then no, wouldn't want to include. It won't change the time remaining... just make for more (smaller) iterations.